pkg/apis/devops/v1alpha2/register.go

```diff
@@ -517,6 +517,130 @@ func addWebService(c *restful.Container) error {
 		Returns(http.StatusOK, RespOK, devops.Crumb{}).
 		Writes(devops.Crumb{}))
 
+	// match /job/init-job/descriptorByName/org.jenkinsci.plugins.workflow.cps.CpsFlowDefinition/checkScriptCompile
+	webservice.Route(webservice.POST("/devops/check/scriptcompile").
+		To(devopsapi.CheckScriptCompile).
+		Metadata(restfulspec.KeyOpenAPITags, tags).
+		Consumes("application/x-www-form-urlencoded", "charset=utf-8").
+		Produces("application/json", "charset=utf-8").
+		Doc("Check pipeline script compile.").
+		Reads("value=\"\"").
+		Returns(http.StatusOK, RespOK, devops.CheckScript{}).
+		Writes(devops.CheckScript{}))
```

Comment:
I recommend returning an array, although it is a single error most of the time.

Reply:
yes, returning an array would be better if the result is failure, but if the result is successful, jenkins will return a map, not array. So for structural unity, i think return a struct will be better